compiler/luci-interpreter/src/kernels/LQFullyConnected.cpp

```diff
@@ -0,0 +1,214 @@
+                                   const Tensor *bias, Tensor *output,
+                                   const LQFullyConnectedParams &params)
+  : KernelWithParams<LQFullyConnectedParams>(
+      {input, input_scales, weights_scales, weights_binary, bias}, {output}, params)
+{
+}
+
+void LQFullyConnected::configure()
+{
+  LUCI_INTERPRETER_CHECK(input()->element_type() == DataType::FLOAT32);
+  LUCI_INTERPRETER_CHECK(input_scales()->element_type() == DataType::FLOAT32);
+  LUCI_INTERPRETER_CHECK(weights_scales()->element_type() == DataType::FLOAT32);
+  LUCI_INTERPRETER_CHECK(weights_binary()->element_type() == DataType::S32);
+  LUCI_INTERPRETER_CHECK(output()->element_type() == DataType::FLOAT32);
+  LUCI_INTERPRETER_CHECK(!bias() || bias()->element_type() == DataType::FLOAT32);
+
+  // encoding lens should be less than 32
+  LUCI_INTERPRETER_CHECK(input_scales()->shape().dim(0) < 32);
+  LUCI_INTERPRETER_CHECK(weights_scales()->shape().dim(1) < 32);
+
+  const Shape &input_shape = input()->shape();
+  const Shape &weights_shape = weights_binary()->shape();
+
+  LUCI_INTERPRETER_CHECK(weights_shape.num_dims() == 3);
+  LUCI_INTERPRETER_CHECK(bias() == nullptr ||
+                         bias()->shape().num_elements() == weights_shape.dim(0));
+
+  const int32_t batch_size = input_shape.dim(0);
+  const int32_t num_units = weights_shape.dim(0);
+  output()->resize({batch_size, num_units});
+
+  // temp binary
+  const uint32_t input_encode_bits = input_scales()->shape().dim(0);
+  const uint32_t binary_width = weights_shape.dim(2);
+
+  LUCI_INTERPRETER_CHECK(binary_width == ceil_div(params().hidden_size, 32));
+
+  input_binary = std::make_unique<QuantizedBinaryInput>(input_shape.dim(1), input_scales());
+}
+
+void LQFullyConnected::execute() const
+{
+  switch (input()->element_type())
+  {
+    case DataType::FLOAT32:
+      evalFloat();
+      break;
+    default:
+      throw std::runtime_error("Unsupported type.");
+  }
+}
+
+void LQFullyConnected::evalFloat() const
+{
+  const float *input_data = getTensorData<float>(input());
+  const float *input_scales_data = getTensorData<float>(input_scales());
+  const float *weight_scales_data = getTensorData<float>(weights_scales());
+  const int32_t *input_binary_data = input_binary->data();
+  const int32_t *weights_binary_data = getTensorData<int32_t>(weights_binary());
+  const float *bias_data = getTensorData<float>(bias());
+  float *output_data = getTensorData<float>(output());
+
+  const int32_t input_encode_bits = input_scales()->shape().dim(0);
+  const int32_t weights_encode_bits = weights_scales()->shape().dim(1);
+
+  const Shape &in_shape = input()->shape();
+  const Shape &out_shape = output()->shape();
+  const int32_t batches = out_shape.dim(0);
+  const int32_t output_size = out_shape.dim(1);
+
+  const int32_t hidden_size = params().hidden_size;
+  const int32_t real_size = weights_binary()->shape().dim(2);
+
+  // TODO implement different versions of this function
```

Comment:
In cases, when 0 bit of encoding denotes `0` instead `-1` we need to use other implementations. Let's solve this problem latter if needed!